File: lib/java/src/main/java/uapi/internal/validation/ValidateResult.java
package uapi.internal.validation;

import static uapi.internal.validation.MapValidationFailuresToInvalidFieldCases.mapValidationFailuresToInvalidFieldCases;

import java.util.List;

import uapi.UApiError;
import uapi.internal.types.UUnion;

public class ValidateResult {
    public static void validateResult(UUnion resultUnionType, Object errorResult) {
        final var newErrorResultValidationFailures = resultUnionType.validate(
                errorResult, List.of(), new ValidateContext(null, null));
        if (!newErrorResultValidationFailures.isEmpty()) {
            throw new UApiError(
                    "Failed internal uAPI validation: "
                            + mapValidationFailuresToInvalidFieldCases(newErrorResultValidationFailures));
        }
    }
}

File: lib/py/uapi/internal/validation/ValidateResult.py
from typing import TYPE_CHECKING
from uapi.UApiError import UApiError

if TYPE_CHECKING:
    from uapi.internal.types.UUnion import UUnion


def validate_result(result_union_type: 'UUnion', error_result: object) -> None:
    from uapi.internal.validation.MapValidationFailuresToInvalidFieldCases import map_validation_failures_to_invalid_field_cases

    new_error_result_validation_failures = result_union_type.validate(
        error_result, None, None, []
    )
    if new_error_result_validation_failures:
        raise UApiError(
            f"Failed internal uAPI validation: {map_validation_failures_to_invalid_field_cases(new_error_result_validation_failures)}"
        )
